api/__tests__/index.test.js

```diff
@@ -0,0 +1,273 @@
+      request(app)
+        .get('/v1/credit-cards/1')
+        .set('Accept', 'application/json')
+        .expect('Content-Type', /json/)
+        .expect(res => {
+          expect(res.body).toEqual({
+            id: 1,
+            name: 'Scotia Momentum Visa Infinitie',
+            interest_rate: '19.99%',
+            annual_fee: '$99',
+            type_id: 1
+          });
+        })
+        .expect(200, done);
+    });
+
+    it("Should respond with 404 when requested item doesn't exist", done => {
+      request(app)
+        .get('/v1/credit-cards/9999')
+        .set('Accept', 'application/json')
+        .expect('Content-Type', /json/)
+        .expect(404, done);
+    });
+  });
+});
+
+describe('Customers Endpoint', () => {
+  describe('/v1/customers', () => {
+    it('Should respond to GET method', done => {
+      request(app)
+        .get('/v1/interests')
+        .set('Accept', 'application/json')
+        .expect('Content-Type', /json/)
+        .expect(200, done);
+    });
+  });
+
+  describe('/v1/customers/:customerId', () => {
+    it('Should respond to GET method', done => {
+      request(app)
+        .get('/v1/customers/1')
+        .set('Accept', 'application/json')
+        .expect('Content-Type', /json/)
+        .expect(res => {
+          expect(res.body.id).toEqual(1);
+        })
+        .expect(200, done);
+    });
+
+    it("Should respond with 404 when requested item doesn't exist", done => {
+      request(app)
+        .get('/v1/customers/9999')
+        .set('Accept', 'application/json')
+        .expect('Content-Type', /json/)
+        .expect(404, done);
+    });
+  });
+
+  describe('/:customerId/profile', () => {
+    it('Should respond to GET method', done => {
+      return request(app)
+        .get('/v1/customers/1/profile')
+        .set('Accept', 'application/json')
+        .expect(res => {
+          expect(res.body[0].customer_id).toEqual(1);
+        })
+        .expect(200, done);
+    });
+  });
+
+  describe('/:customerId/interests', () => {
+    it('Should respond to GET method', done => {
+      return request(app)
+        .get('/v1/customers/1/interests')
+        .set('Accept', 'application/json')
+        .expect(res => {
+          expect(res.body[0].name).toEqual("Traveling");
+        })
+        .expect(200, done);
+    });
+  });
+
+  describe('/:customerId/transactions', () => {
+    it('Should respond to GET method', done => {
+      return request(app)
+        .get('/v1/customers/1/transactions')
+        .set('Accept', 'application/json')
+        .expect(res => {
+          expect(res.body[0].customer_id).toEqual(1);
+        })
+        .expect(200, done);
+    });
+  });
+
+  describe('/:customerId/balances', () => {
+    it('Should respond to GET method', done => {
+      return request(app)
+        .get('/v1/customers/1/balances')
+        .set('Accept', 'application/json')
+        .expect(res => {
+          expect(res.body[0].name).toEqual('Air Miles');
```

Comment:
As a sidenote, if the data was mocked, this would have been a good assertion. Since this is using the real DB, a more proper assertion would be to check if it is not null or empty.